spec/rdkafka/producer_spec.rb

```diff
@@ -407,4 +407,23 @@
     # Waiting a second time should work
     handle.wait(max_wait_timeout: 5)
   end
+
+  def finalizer_proc
+    # The proc will hold a reference to the producer when defined in the same scope.
+    # https://www.mikeperham.com/2010/02/24/the-trouble-with-ruby-finalizers/
+    proc { @cleared = true }
+  end
+
+  it "should close open producer sockets before they are GC'd" do
+    skip if defined?(JRUBY_VERSION)
+
+    @cleared = false
+    producer = rdkafka_config.producer
+    ObjectSpace.define_finalizer(producer, finalizer_proc)
+    producer = nil
+    # Request GC run.
+    RUBY_PLATFORM.casecmp?('java') ? java.lang.System.gc : GC.start
```

Comment:
GC run is not guaranteed upon invocation as per the docs, so this test could be unreliable as-is.

The approach to validate the instance variable is not working in JRuby even though the finalizer is called.
https://github.com/jruby/jruby/issues/6353